roles/varnish/templates/etc/varnish/varnish.vcl

```diff
@@ -269,31 +254,39 @@ sub vcl_recv {
         }
         elsif (req.restarts == 0  && req.url ~ "^/content/col.*/?\?format=rdf") {
             set req.backend_hint = legacy_frontend;
-            return(hash);
         }
         elsif (req.restarts == 0  && req.url ~ "^/content/.*/module_export_template") {
             set req.backend_hint = legacy_frontend;
-            return(hash);
         }
         elsif (req.restarts == 0  && req.url ~ "^/content/(m[0-9]+)/([0-9.]+)/.*format=pdf$") {
             set req.backend_hint = static_files;
             set req.url = regsub(req.url, "^/content/(m[0-9]+)/([0-9.]+)/.*format=pdf", "/files/\1-\2.pdf");
+            return (pipe);
```

Comment:
Hmm, `pipe` is pretty strongly discouraged in varnish-lore, especially for what are actual HTTP transactions. I recall us having problems w/ not getting the last few kbytes of some files sometimes, solved by removing some pipes.

Oh btw, this restarts code is an example for the earlier comment - it tries to get the module PDFs from nginx, and if that fails, calls zope, which builds the PDF, serves it, _and_ stores it on the filesystem, for the next call.

Reply:
I think you are right. Though piping to nginx would be ideal, the restart may fail if we pipe here. I'll take it out.

Reply:
Actually... even though this code pretends to handle restarts... the only restart that would ever happen would be on `/files/`, since that's the only route that can `return (retry)`